sycl/source/detail/device_image_impl.hpp

```diff
@@ -220,6 +220,10 @@ class device_image_impl {
       const detail::plugin &Plugin = getSyclObjImpl(MContext)->getPlugin();
       Plugin.call<PiApiKind::piProgramRelease>(MProgram);
     }
+    if (MSpecConstsBuffer) {
```

Comment:
Should the `MSpecConstAccessMtx` lock be held while checking that `MSpecConstsBuffer` exists and freeing it?

Reply:
Yes, it's better to have it. 'MSpecConstAccessMtx' added.